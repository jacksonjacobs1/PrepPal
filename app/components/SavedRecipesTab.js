import React, { useEffect, useState } from 'react';
import { StyleSheet, ScrollView, Text, View, FlatList, SafeAreaView } from 'react-native';
import { SearchBar } from 'react-native-elements';
import { List } from 'react-native-paper';
import { getUserRecipes } from '../firebaseApp';
import { getAuth } from '@firebase/auth';

function SavedRecipesTab({ navigation }) {
    const auth = getAuth();
    const user = auth.currentUser;
    const [data, setData] = useState([])
    const [searchText, setSearchText] = useState('')

    useEffect(() => {
        getUserRecipes(user).then(response => { setData(response) })
    }, [])

    if (!data.length) return (<Text>Loading...</Text>);

    return (
        <View>
            <SearchBar
                placeholder="Search recipes"
                onChangeText={searchText => setSearchText(searchText)}
                value={searchText}
            />
            <ScrollView>
            <List.AccordionGroup>
                {
                    data.map(item => (
                        <List.Accordion key={item.uid} title={item.name} id={item.uid}>
                            <View>
                                <List.Item title='Ingredients' description={item.ingredients}  />
                                <List.Item title='Instructions' description={item.instructions} />
                            </View>
                        </List.Accordion>
                    ))
                }
            </List.AccordionGroup>
        </ScrollView>
        </View>
        

    );
}

const styles = StyleSheet.create({
    savedRecipesTitle: {
        marginTop: 0,
        paddingVertical: 6,
        borderWidth: 3,
        borderColor: "#20232a",
        color: "black",
        backgroundColor: "green",
        fontSize: 20,
        fontWeight: "bold",
        borderRadius: 6,
        textAlign: "center"
    },
});

export default SavedRecipesTab;