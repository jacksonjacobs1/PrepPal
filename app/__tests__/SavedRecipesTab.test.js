import React from "react";
import renderer from 'react-test-renderer';
import SavedRecipesTab from "../components/SavedRecipesTab";

it('renders correctly', () => {
  const tree = renderer.create(<SavedRecipesTab />).toJSON();
  expect(tree).toMatchSnapshot();
});