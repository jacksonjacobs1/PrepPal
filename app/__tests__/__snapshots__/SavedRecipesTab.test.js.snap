// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  testID="savedRecipesTab"
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#393e42",
          "borderBottomColor": "#000",
          "borderBottomWidth": 1,
          "borderTopColor": "#000",
          "borderTopWidth": 1,
          "padding": 8,
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 0,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#303337",
              "borderBottomWidth": 0,
              "borderColor": "#86939e",
              "borderRadius": 3,
              "flexDirection": "row",
              "minHeight": 30,
              "overflow": "hidden",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 8,
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Search recipes"
            placeholderTextColor="#86939e"
            platform="default"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#86939e",
                "flex": 1,
                "fontSize": 18,
                "marginLeft": 10,
                "minHeight": 40,
              }
            }
            testID="savedSearch"
            underlineColorAndroid="transparent"
            value=""
          >
             
          </TextInput>
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginRight": 8,
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            />
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "height": 0,
              "margin": 0,
              "padding": 0,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "backgroundColor": "green",
          "borderColor": "#20232a",
          "borderRadius": 6,
          "borderWidth": 3,
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginTop": 0,
          "paddingVertical": 6,
          "textAlign": "center",
        }
      }
    >
       Your saved recipes: 
    </Text>
    <Text>
      ⬤
       Recipe 1
    </Text>
    <Text>
      ⬤
       Recipe 2
    </Text>
    <Text>
      ⬤
       Recipe 3
    </Text>
    <Text>
      ⬤
       Recipe 4
    </Text>
    <Text>
      ⬤
       Recipe 5
    </Text>
    <Text>
       
    </Text>
    <Text>
      (Note: the above search bar is still under construction.)
    </Text>
  </View>
</RCTScrollView>
`;
